src/main/java/dev/foxikle/customnpcs/internal/listeners/Listeners.java

```diff
@@ -79,88 +206,75 @@ public void onPlayerInteract(PlayerInteractEntityEvent e) {
-            });
-            e.setCancelled(true);
-        } else if (plugin.messageWaiting.contains(e.getPlayer())) {
-            Bukkit.getScheduler().runTask(plugin, () -> {
-                plugin.messageWaiting.remove(e.getPlayer());
-                Action action = plugin.editingActions.get(e.getPlayer());
-                List<String> currentArgs = action.getArgs();
-                currentArgs.clear();
-                currentArgs.addAll(List.of(e.getMessage().split(" ")));
-                e.getPlayer().sendMessage(Component.text("Successfully set message to be '", NamedTextColor.GREEN).append(plugin.getMiniMessage().deserialize(e.getMessage())).append(Component.text("'", NamedTextColor.GREEN)));
-                Bukkit.getScheduler().runTask(plugin, () -> e.getPlayer().openInventory(plugin.menuCores.get(e.getPlayer()).getActionCustomizerMenu(action)));
-            });
-            e.setCancelled(true);
-        } else if (plugin.serverWaiting.contains(e.getPlayer())) {
-            Bukkit.getScheduler().runTask(plugin, () -> {
-                plugin.serverWaiting.remove(e.getPlayer());
-                Action action = plugin.editingActions.get(e.getPlayer());
-                List<String> currentArgs = action.getArgs();
-                currentArgs.clear();
-                currentArgs.addAll(List.of(e.getMessage().split(" ")));
-                e.getPlayer().sendMessage(ChatColor.GREEN + "Successfully set server to be '" + ChatColor.RESET +  e.getMessage() + ChatColor.RESET + "" + ChatColor.GREEN + "'");
-                Bukkit.getScheduler().runTask(plugin, () -> e.getPlayer().openInventory(plugin.menuCores.get(e.getPlayer()).getActionCustomizerMenu(action)));
-            });
-            e.setCancelled(true);
-        } else if (plugin.actionbarWaiting.contains(e.getPlayer())) {
-            Bukkit.getScheduler().runTask(plugin, () -> {
-                plugin.actionbarWaiting.remove(e.getPlayer());
-                Action action = plugin.editingActions.get(e.getPlayer());
-                List<String> currentArgs = action.getArgs();
-                currentArgs.clear();
-                currentArgs.addAll(List.of(e.getMessage().split(" ")));
-                e.getPlayer().sendMessage(Component.text("Successfully set actionbar to be '", NamedTextColor.GREEN).append(plugin.getMiniMessage().deserialize(e.getMessage())).append(Component.text("'", NamedTextColor.GREEN)));
-                Bukkit.getScheduler().runTask(plugin, () -> e.getPlayer().openInventory(plugin.menuCores.get(e.getPlayer()).getActionCustomizerMenu(action)));
-            });
-            e.setCancelled(true);
+            plugin.targetWaiting.remove(player);
+            conditional.setTargetValue(message);
+            player.sendMessage(ChatColor.translateAlternateColorCodes("&aSuccessfully set target to be '&r" + message + "&a'"));
```

Comment:
Translation looks nicer, but you can skip it by using the `§` symbol and the letter